Add ability to pass errors out of cache to UI

ledbettj opened the pull request:
* Renamed `TimedString` to `CachedBytes`
* Updated `CachedBytes` to indicate if it failed to refresh itself and return an error.
* Updated the endpoint to return an error if the cache is invalid and refresh failed; currently this is just plain text, but it probably needs to be JSON for the frontend to handle it correctly.   Currently we report an error if any `reqwest` operation fails, if MARTA returns a 500 or 401, or any non-200 status code.

ledbettj committed:
rename TimedString to CachedBytes

ledbettj committed:
Giant overengineering to add error support

ledbettj commented on src/cached_bytes.rs:
```diff
@@ -0,0 +1,129 @@
+
+impl CachedBytes {
+    pub fn new<S: Into<String>>(url: S, ttl: Duration) -> CachedBytes {
+        CachedBytes::new_with_updater(url, ttl, |endpoint| {
```
new default update function got a little more complicated; it now does error checking against what we get back from the URL.

jakswa commented:
this might be a good time to undo my removal of the `serde` helpers, and have this new object be easy to turn into JSON when an error occurs. I The warp repo had [one example](https://github.com/seanmonstar/warp/blob/master/examples/todos.rs#L18) of this I saw, but maybe there are others.

Can always come back through later and do that though. We're in ok-to-ignore conversation territory now. The main motivation is eventually bringing back the react conditionals that say stuff like, "marta is probably broken, not my fault" if it's the known cases we've had in the past. This has been done differently in my previous iterations and we're free to make our own way:
- elixir pattern https://github.com/jakswa/miotwo/commit/e8ed528f4199d720bf63283fdc384a638c199938
- really old nodejs patterns https://github.com/jakswa/marta_api/commit/d33ed59ef1c3a7d5122b8b1849325543c1fc0a4a
- discussion and UI work on the old ionic repo https://github.com/jakswa/martaionic/issues/2

jakswa committed:
also handle is_timeout() possibility

get() might return an Err representing various client errors that don't
result in a Response. Timeout is one of those and I got to test that
tonight. When one of these client errors happen we'll return a generic
500, except for the timeout case.

jakswa commented:
Got to test this tonight on marta's API as it was timing out. Even though it didn't cover the "timeout" case explicitly, the `get()?` behavior of automatically returning the error early saved things. It was just a little too open about things for our case:
![image](https://user-images.githubusercontent.com/137793/67357714-22a33c00-f52c-11e9-8769-a27567b0f063.png)

I don't have the timeout case to test anymore, but I followed docs and I think I got 868dc3d doing the right thing (comments welcome though, it was a struggle to keep things clean...).
